src/modules/SdnDiag.Health/private/Test-ResourceConfigurationState.ps1

```diff
@@ -4,24 +4,39 @@ function Test-ResourceConfigurationState {
         Validate that the configurationState of the resources.
     #>
 
-    [CmdletBinding()]
+    [CmdletBinding(DefaultParameterSetName = 'RestCredential')]
     param (
         [Parameter(Mandatory = $true)]
         [SdnFabricEnvObject]$SdnEnvironmentObject,
 
-        [Parameter(Mandatory = $false)]
+        [Parameter(Mandatory = $false, ParameterSetName = 'RestCredential')]
         [System.Management.Automation.PSCredential]
         [System.Management.Automation.Credential()]
-        $NcRestCredential = [System.Management.Automation.PSCredential]::Empty
+        $NcRestCredential = [System.Management.Automation.PSCredential]::Empty,
+
+        [Parameter(Mandatory = $true, ParameterSetName = 'RestCertificate')]
```

Comment:
Review `Mandatory` attribute

Reply:
This is correct when using ParameterSetName and helps define what is required. It enables unique properties to be required depending on the ParameterSetName that is used.